src/uid2ApiClient.ts

```diff
@@ -156,4 +189,91 @@ export class Uid2ApiClient {
     return promise;
   }
+
+  public async callCstgApi(
+    data: { emailHash: string } | { phoneHash: string },
+    subscriptionId: string,
+    serverPublicKey: string
+  ): Promise<CstgResult> {
+    const request =
+      "emailHash" in data
+        ? { email_hash: data.emailHash }
+        : { phone_hash: data.phoneHash };
+
+    const box = await UID2CstgBox.build(serverPublicKey);
+
+    const encoder = new TextEncoder();
+
+    const now = Date.now();
+    const { iv, ciphertext } = await box.encrypt(
+      encoder.encode(JSON.stringify(request)),
+      encoder.encode(JSON.stringify([now]))
+    );
+
+    const exportedPublicKey = await exportPublicKey(box.clientPublicKey);
+
+    const requestBody = {
+      payload: Uid2ApiClient.bytesToBase64(new Uint8Array(ciphertext)),
+      iv: Uid2ApiClient.bytesToBase64(new Uint8Array(iv)),
+      public_key: Uid2ApiClient.bytesToBase64(
+        new Uint8Array(exportedPublicKey)
+      ),
+      timestamp: now,
+      subscription_id: subscriptionId,
+    };
+
+    const url = this._baseUrl + "/v2/token/client-generate";
+    const req = new XMLHttpRequest();
+    this._requestsInFlight.push(req);
+    req.overrideMimeType("text/plain");
+    req.open("POST", url, true);
+    req.setRequestHeader("X-UID2-Client-Version", this._clientVersion);
+
+    let resolvePromise: (result: CstgResult) => void;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let rejectPromise: (reason?: any) => void;
+    const promise = new Promise<CstgResult>((resolve, reject) => {
+      resolvePromise = resolve;
+      rejectPromise = reject;
+    });
+
+    req.onreadystatechange = async () => {
+      if (req.readyState !== req.DONE) return;
+      this._requestsInFlight = this._requestsInFlight.filter((r) => r !== req);
+      try {
+        if (req.status !== 200) {
+          const response = JSON.parse(req.responseText) as unknown;
+          const result = this.ResponseToCstgResult(response);
+          if (typeof result === "string") rejectPromise(result);
+          else resolvePromise(result);
+        } else {
+          const encodedResp = this.createArrayBuffer(atob(req.responseText));
+          const decrypted = await box.decrypt(
+            encodedResp.slice(0, 12),
+            encodedResp.slice(12)
+          );
+          const decryptedResponse = new TextDecoder().decode(decrypted);
+          const response = JSON.parse(decryptedResponse) as unknown;
+          const result = this.ResponseToCstgResult(response);
+          if (typeof result === "string") rejectPromise(result);
+          else resolvePromise(result);
+        }
+      } catch (err) {
+        // TODO: Log error? See callRefreshApi.
+        rejectPromise(err);
+      }
+    };
+
+    req.send(JSON.stringify(requestBody));
+
+    return await promise;
+  }
+
+  // TODO: See if this is already implemented elsewhere.
+  private static bytesToBase64(bytes: Uint8Array): string {
```

Comment:
I notice this method is implemented here and in `uid2Sdk.ts`. Would it be a good idea to move this to somewhere shared?